device/thunder_ripple_sdk/src/processors/panel/thunder_hdmi.rs

```diff
@@ -0,0 +1,371 @@
+                panel::device_hdmi::HdmiRequest,
+            },
+            firebolt::panel::fb_hdmi::{GetAvailableInputsResponse, HdmiSelectOperationResponse},
+        },
+        async_trait::async_trait,
+        extn::{
+            client::extn_client::ExtnClient,
+            client::extn_processor::{
+                DefaultExtnStreamer, ExtnRequestProcessor, ExtnStreamProcessor, ExtnStreamer,
+            },
+            extn_client_message::{ExtnMessage, ExtnResponse},
+        },
+        log::info,
+        serde_json,
+        tokio::sync::mpsc,
+        utils::error::RippleError,
+    },
+    thunder_state::ThunderState,
+};
+use ripple_sdk::{
+    api::{
+        apps::{AppEvent, AppEventRequest},
+        device::device_events::DeviceEventCallback,
+        firebolt::panel::fb_hdmi::{
+            AutoLowLatencyModeSignalChangedInfo, HdmiConnectionChangedInfo,
+            HdmiSelectOperationRequest,
+        },
+    },
+    extn::extn_client_message::ExtnEvent,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug)]
+pub struct ThunderHdmiRequestProcessor {
+    state: ThunderState,
+    streamer: DefaultExtnStreamer,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct AVInputGetInputDevicesParams {
+    type_of_input: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct AVInputStartHdmiInputParams {
+    port_id: u32,
+    type_of_input: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct AVInputStopHdmiInputParams {
+    type_of_input: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ThunderHdmiConnectionChanged {
+    pub id: u32,
+    pub locator: String,
+    pub connected: bool,
+}
+
+impl From<ThunderHdmiConnectionChanged> for HdmiConnectionChangedInfo {
+    fn from(value: ThunderHdmiConnectionChanged) -> Self {
+        HdmiConnectionChangedInfo {
+            port: format!("HDMI{}", value.id),
+            connected: value.connected,
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ThunderAutoLowLatencyModeSignalChanged {
+    pub id: u32,
+    pub game_feature: String,
+    pub mode: bool,
+}
+
+impl From<ThunderAutoLowLatencyModeSignalChanged> for AutoLowLatencyModeSignalChangedInfo {
+    fn from(value: ThunderAutoLowLatencyModeSignalChanged) -> Self {
+        AutoLowLatencyModeSignalChangedInfo {
+            port: format!("HDMI{}", value.id),
+            auto_low_latency_mode_signalled: value.mode,
+        }
+    }
+}
+
+impl ThunderHdmiRequestProcessor {
+    pub fn new(state: ThunderState) -> ThunderHdmiRequestProcessor {
+        ThunderHdmiRequestProcessor {
+            state,
+            streamer: DefaultExtnStreamer::new(),
+        }
+    }
+
+    async fn get_available_inputs(state: ThunderState, req: ExtnMessage) -> bool {
```

Comment:
All of the handlers look basically the same. Is there a way we can reduce the repeated code?

Reply:
Certainly Start and Stop can be just one value in the operation which is what was suggested. Get available inputs body is slightly different so there would be something additional. 

Think we can move the Thunder call alone to a separate function to reduce the boilerplate

Reply:
Refactored it in latest code.